### ISSUE TITLE
get_total_results() is slowing whole process

### ISSUE THREAD

**Issue by digitaloutback** [IgnitedDatatables/Ignited-Datatables#82]
By default, get_total_results() does a ci->db->get() which runs a SELECT *

Without hacking source code, is there a way to choose the column which it is using. My table has 50,000 results and it goes from a 1.8 second query to 0.2 if I change the select clause to the first column of my table.

Any suggested workarounds without hacking? (I am using the library on several areas of the project)

**Comment by olarriga** [comment]
I have the same problem ... but worse. The query that counts the number of rows takes 2s and generates 200MB of traffic while the query to display the lines only takes 1ms.

**Comment by digitaloutback** [comment]
I overcame this by editing Datatables.php:

1. A new private variable:
```
private $count_column;
```

2. A new function within the class:
```    
    /**
    * WRA added function to set the count column
    *
    * @param string $column
    * @return mixed
    */
    public function set_count_column($column)
    {
      $this->count_column = $column;
      return $this;
    }
```

3. And this in the get_total_results() function around line 436:
```
      if(strlen($this->count_column) > 0)
        $this->ci->db->select($this->count_column);
```

4. And then in my controller:
```
$this->datatables->set_count_column('id'); 
```

**Comment by olarriga** [comment]
I replaced the last 2 lines of the get_total_results() by this line:
return $this->ci->db->count_all_results($this->table);

It looks run.

I will also try your solution. Thanks.

**Comment by olimortimer** [comment]
@digitaloutback that sounds quite useful.  Are you planning to submit it as a pull request?

**Comment by zazikhan** [comment]
I am new to datatables i have 94,000 rows and it take 8 to 9 secs to get loaded i think the problem is get_total_record function so what should i do make my records get fetched faster

**Comment by digitaloutback** [comment]
Yes just folliw the instructions that I wrote. Also try to switch to serverside processing setting. Remember to also ensure any table join fields are indexed.

**Comment by zazikhan** [comment]
Let me explain fully i have server side enable **"bServerSide"=true,** also i tried your code which didnt work it was giving me the same result. than for testing purpose i modified my datatable library to retrieve only 10 records and nothing happened let me share my change to make it more clear .

`  private function produce_output($output, $charset)
    {
      $aaData = array();
      $rResult = $this->get_display_result();

      if($output == 'json')
      {
        $iTotal = 10;//$this->get_total_results();
        $iFilteredTotal = 10;//$this->get_total_results(TRUE);
      }`

so i disable the count functionality and i put static values so i understand one thing that problem isnt with this function $this->get_total_results();

**Comment by pablogomezca** [comment]
These solutions works only if your are not using filtering.
